mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m

```diff
@@ -0,0 +1,703 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "mediapipe/tasks/ios/common/sources/MPPCommon.h"
+#import "mediapipe/tasks/ios/test/vision/utils/sources/MPPImage+TestUtils.h"
+#import "mediapipe/tasks/ios/vision/object_detector/sources/MPPObjectDetector.h"
+
+static NSString *const kModelName = @"coco_ssd_mobilenet_v1_1.0_quant_2018_06_29";
+static NSDictionary *const kCatsAndDogsImage = @{@"name" : @"cats_and_dogs", @"type" : @"jpg"};
+static NSDictionary *const kCatsAndDogsRotatedImage =
+    @{@"name" : @"cats_and_dogs_rotated", @"type" : @"jpg"};
+static NSString *const kExpectedErrorDomain = @"com.google.mediapipe.tasks";
+
+#define PixelDifferenceTolerance 5.0f
+#define ScoreDifferenceTolerance 1e-2f
+
+#define AssertEqualErrors(error, expectedError)                                               \
```

Comment:
I think these macros are getting a bit more complicated than what I would think is acceptable for macros. Can we turn this into helper functions?

Reply:
I have converted them into helper functions which use macros only for comparison of two categories and bounding boxes.